app/javascript/packs/shared/reducers/TeamReducers.js

```diff
@@ -0,0 +1,16 @@
+import { SET_CURRENT_TEAM, GET_LIST_OF_TEAMS } from "../actions/types";
+
+let initial_state = { name: "", id: 0, current_team:true }
+export const setCurrentTeam = (state = initial_state, action) => {
+  if (action.type === SET_CURRENT_TEAM) {
+    return Object.assign({}, state, action.user);
+  }
+  return state;
+};
+
+export const getListOfTeams = (state = [initial_state], action) => {
+  if (action.type === GET_LIST_OF_TEAMS) {
+    return  Object.assign([], state, action.payload);
```

Comment:
Are state and action.payload objects? If they are than you can ignore the comment below. If not then

Object.assign({}, ....) has some weird behaviour if you use array as first argument

`Object.assign([], ["tonko", "dadi"], ["marjan", "alenka"])`
`=> [ 'marjan', 'alenka' ]`